ansible_collections/lvrfrc87/git_acp/plugins/modules/git_acp.py

```diff
@@ -277,17 +286,27 @@ def main():
             )
 
     git = Git(module)
-    changed_files = git.status()
+    # checkout branch if parameter was given
+    if branch:
+        git.checkout()
 
-    if changed_files:
-        if pull:
-            result.update(git.pull())
+    if pull:
+        result.update(git.pull())
+    
+    changed_files = git.status()
 
+    if merge:
+        result.update(git.merge())
+    elif changed_files and add:
         git.add()
         result.update(git.commit())
+    elif tag:
+        result.update(git.tag())
+    else:
+        push = False
 
-        if push:
-            result.update(git.push())
+    if push:
+        result.update(git.push())
         result["changed"] = True
```

Comment:
Is this indentation correct? It seems something got scrambled up here.

Reply:
that's intended

Reply:
Okay! Just make sure "changed" is always set properly, thus set to "true" only when a change is actually occurred.